sphinx/cmd/quickstart.py

```diff
@@ -464,87 +465,146 @@ def valid_dir(d: Dict) -> bool:
     return True
 
 
-def get_parser() -> argparse.ArgumentParser:
-    description = __(
-        "\n"
-        "Generate required files for a Sphinx project.\n"
-        "\n"
-        "sphinx-quickstart is an interactive tool that asks some questions about your\n"
-        "project and then generates a complete documentation directory and sample\n"
-        "Makefile to be used with sphinx-build.\n"
-    )
-    parser = argparse.ArgumentParser(
-        usage='%(prog)s [OPTIONS] <PROJECT_DIR>',
-        epilog=__("For more information, visit <http://sphinx-doc.org/>."),
-        description=description)
+class InitCommand(SphinxCommand):
+    """Generate required files for a Sphinx project.
 
-    parser.add_argument('-q', '--quiet', action='store_true', dest='quiet',
-                        default=None,
-                        help=__('quiet mode'))
-    parser.add_argument('--version', action='version', dest='show_version',
-                        version='%%(prog)s %s' % __display_version__)
+    'sphinx init' is an interactive tool that asks some questions about your
```

Comment:
In some tools, `init` subcommand goes project initialization silently. So I wonder we should separate quickstart feature into `sphinx init` and `sphinx quickstart` (?) (or `sphinx init --interactive`).

Reply:
Are you thinking of `git init`? I think it's okay for `sphinx init` to be interactive, so long as you can specify these options from the command line and avoid the interactive prompts. We do need some way to retrieve required information like `project` and `author`.

Reply:
Yes, I was thinking of `git init` and `bundle init`. But I found `npm init` asks questions interactively.  So please forget my comment.

>We do need some way to retrieve required information like project and author.

Actually, `project` and `author` are not required configuration. So we can suppress all of questions.

Reply:
> Actually, `project` and `author` are not required configuration. So we can suppress all of questions.

Oh, I didn't know that. Good to know. However, is it really useful to ignore these? What advantage would `git init` provide over `touch conf.py` if we didn't ask for these? I think it makes sense to ask for these at a minimum, even if people can skip past them.

Reply:
It seems some users would like to do interactions more verbosely (see #6656). I still don't have an answer for it. But, as one of idea, it might be better to provide both silent and verbose way.

Note: silent means only "less". It does not equal to no question.

Reply:
Sorry for the delay. How about we make interactive *not the default* by adding an `--interactive` flag? With this flag, we would request *all* configuration options in an interactive manner from the user, but users that don't want to do this can simply pass the relevant args they care about, e.g. `--project`? That would satisfy the request of #6656 while not annoying people like you that just wants to start a really quick project? The other alternative is a combination of `--quick` (default) and `--full` arguments, with the latter enabling configuration of extensions etc. What do you think?

Reply:
I prefer an option like `--full`. It is too much to choose non interactive mode by default to me. But it is annoying to me to list all configurations. I think it is difficult to choose to appropriate options or extensions for standard users. So it would be better to ask minimum configurations by default.